client/file-manager/src/rocksdb.rs

```diff
@@ -780,19 +844,35 @@ where
             &chunk_count.to_le_bytes(),
         );
 
-        let bucket_prefixed_file_key = metadata
-            .bucket_id()
-            .iter()
-            .copied()
-            .chain(file_key.as_ref().iter().copied())
-            .collect::<Vec<_>>();
+        let bucket_prefixed_file_key = Self::build_bucket_prefixed_file_key(&metadata, &file_key);
 
         // Store the key prefixed by bucket id
         transaction.put(
             Column::BucketPrefix.into(),
             bucket_prefixed_file_key.as_ref(),
             &[],
         );
+        // Increment fingerprint refcount
+        let current = self
+            .storage
+            .read(
+                Column::FingerprintRefCount.into(),
+                metadata.fingerprint().as_ref(),
+            )
+            .map_err(|e| {
+                error!(target: LOG_TARGET, "{:?}", e);
+                FileStorageError::FailedToReadStorage
+            })?
+            .map(|bytes| u64::from_le_bytes(bytes.try_into().unwrap()))
+            .unwrap_or(0);
+        let new_count = current
+            .checked_add(1)
+            .ok_or(FileStorageError::FailedToWriteToStorage)?;
+        transaction.put(
+            Column::FingerprintRefCount.into(),
+            metadata.fingerprint().as_ref(),
+            &new_count.to_le_bytes(),
+        );
```

Comment:
I think we could have a common function for either decreasing or increasing the refcount of some file key and it returning the resultant value